src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,55 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> element;
```

Comment:
```suggestion
    private Node<T> top;
```
Let's rename this field to "top" or "upper".

Reply:
I want to have as it is (: